tests/func/test_remote.py

```diff
@@ -253,3 +255,21 @@ def unreliable_upload(self, from_file, to_info, name=None, **kwargs):
 def get_last_exc(caplog):
     _, exc, _ = caplog.records[-2].exc_info
     return exc
+
+
+def test_raise_on_too_many_open_files(tmp_dir, dvc, tmp_path_factory, mocker):
+    storage = tmp_path_factory.mktemp("test_remote_base")
+    remote_config = RemoteConfig(dvc.config)
+    remote_config.add("local_remote", fspath(storage), default=True)
+
+    tmp_dir.dvc_gen({"file": "file content"})
+
+    too_many_open_files_error = OSError()
+    mocker.patch.object(too_many_open_files_error, "errno", 24)
+    mocker.patch.object(
+        RemoteLOCAL, "_upload", side_effect=too_many_open_files_error
```

Comment:
I'm pretty sure you can just
```suggestion
        RemoteLOCAL, "_upload", side_effect=OSError(24)
```